### PR TITLE
Simplify ObjectAdapter in JS

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR simplifies ObjectAdapter in JS by removing several methods:

- activate
You never need to activate an OA in JS since it can't accept connections. An OS in JS can only be associated with an outgoing connection.

- hold / waitForHold
hold has no effect on outgoing connections (in all languages)

- waitForDeactivate / isDeactivated
these are very uncommon methods, and not all that useful in JS since OA.deactivate is synchronous

### CODE REVIEW

File: js/test/Ice/middleware/Server.ts [hunk 1]
```diff
@@ -75,6 +79,7 @@ export class Server extends TestHelper {
             test(inLog[1] === "B");
             test(inLog[2] === "C");
 
+            out.write("outLog.length = " + outLog.length + "\n");
```
Comment (pepone) [inline js/test/Ice/middleware/Server.ts]: leftover?

File: js/src/Ice/ObjectAdapter.js [hunk 2]
```diff
@@ -570,35 +533,6 @@ export class ObjectAdapter {
     }
 
     setState(state) {
```
Comment (pepone) [inline js/src/Ice/ObjectAdapter.js:535]: Maybe we can just remove this method and just do `this._state = state` in the few places we called it.

**Review by pepone — APPROVED** [review]

File: js/test/Ice/middleware/Server.ts (hunk 1 above)
Comment (bernardnormier) [reply js/test/Ice/middleware/Server.ts]: Yes, fixed.

**Review by externl — APPROVED** [review]

File: js/src/Ice/ObjectAdapter.js (hunk 2 above)
Comment (bernardnormier) [reply js/src/Ice/ObjectAdapter.js]: Fixed as suggested.

**Pull request merged**